### ISSUE TITLE
doxygen can not run in parallel to create multiple man pages (Origin: bugzilla #791153)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#6293]
status NEW severity *normal* in component *general* for *---*
Reported in version *1.8.14-GIT* on platform *Other*
Assigned to: Dimitri van Heesch

On 2017-12-03 06:31:47 +0000, Orivej Desh wrote:

> I'm not sure if this use case is supported, but editorconfig-core-c [1], instantiates multiple copies of Doxygen.in [2] with slightly different parameters to generate man1 and man5 pages separately from other documentation. This may fail at [3] when two processes create the intermediate "man" directory simultaneously while attempting to create "man/man1" and "man/man5" with:
> 
> cmake   | [ 52%] Generating man/man5/editorconfig-format.5
> cmake   | [ 52%] Generating man/man1/editorconfig.1
> doxygen | error: Could not create output directory /tmp/nix-build-editorconfig-core-c-0.12.1.drv-0/editorconfig-core-c-0.12.1/build/doc/man
> 
> This can be mitigated by using QDir.mkpath instead of QDir.mkdir.
> 
> There are other cases of 'if (!exists && !mkdir)' in the doxygen source code, but it is not clear if they need to be fixed.
> 
> [1] https://github.com/editorconfig/editorconfig-core-c/blob/SHA: 008820a0aee25c0cd0383f7d5a2d81241c1f6400/doc/CMakeLists.txt
> [2] https://github.com/editorconfig/editorconfig-core-c/blob/SHA: 008820a0aee25c0cd0383f7d5a2d81241c1f6400/doc/Doxyfile.in
> [3] https://github.com/doxygen/doxygen/blob/SHA: 4f45bd20d4da7d40c793ec4c4c13558581e995ac/src/mangen.cpp#L113

**Comment by orivej** [comment]
The links were broken by the conversion.
Original ticket: https://bugzilla.gnome.org/show_bug.cgi?id=791153
[1] https://github.com/editorconfig/editorconfig-core-c/blob/008820a0aee25c0cd0383f7d5a2d81241c1f6400/doc/CMakeLists.txt
[2] https://github.com/editorconfig/editorconfig-core-c/blob/008820a0aee25c0cd0383f7d5a2d81241c1f6400/doc/Doxyfile.in
[3] https://github.com/doxygen/doxygen/blob/4f45bd20d4da7d40c793ec4c4c13558581e995ac/src/mangen.cpp#L113

**Comment by albert-github** [comment]
Can you please attach a, small, self contained example (source+config file in a tar or zip) that allows us to reproduce the problem instead of the mentioned links?

**Comment by albert-github** [comment]
This usage is not foreseen / supported (as far as I know). 
Always add the information to the question and not to an external link as they might not be persistent.
In the given `CMakeList.txt` I can see that doxygen is ran a couple of times besides for `man1` and `man5` without html also for `man3` with `html`. So in parallel runs it might happen that these runs interfere.
I think a better solution would be either:
- create the output directories beforehand (so the directories `html`, `man/man1`, `man/man3` and `man/man5`) though I'm not sure whether or not you still can run into problems
- create a separate `OUTPUT_DIRECTORY`  for each run and at the end of the run do either:
   - move the relevant directories to the final output directory and install from this final directory
   - install from these temporary directories


I don't see this problem as a doxygen problem, setting it to won't fix
